apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/_components/claim-conditions/claim-conditions-form/hooks.ts

```diff
@@ -175,18 +199,33 @@ function toDate(timestamp: number | Date | undefined) {
   }
   return new Date(timestamp);
 }
-function toBigInt(value: string | number | undefined) {
+function toBigInt(
+  value: string | number | undefined,
+  decimals?: number,
+): bigint | undefined {
   if (value === undefined) {
     return undefined;
   }
   if (value === "unlimited") {
     return maxUint256;
   }
+  // The ERC20Claim condition extension in v5 does not convert to wei for us
+  // so we have to, manually
+  if (decimals) {
+    return toUnits(value.toString(), decimals);
+  }
+
   return BigInt(value);
 }
 
-function toUnlimited(value: bigint) {
-  return value === maxUint256 ? "unlimited" : value.toString();
+function toUnlimited(value: bigint, decimals?: number) {
+  if (value === maxUint256) {
+    return "unlimited";
+  }
+  if (decimals) {
+    return toTokens(value, decimals);
```

Comment:
why do you need this here? unlimited should be the same no matter the decimals no?

Reply:
This is similar to toBigInt. While toBigInt converts the value to send it to the blockchain, toUnlimited converts the value from the blockchain to display it on the UI.

Without this new logic, a maxClaimableSupply of 100 tokens will be displayed on the frontend as: `100000000000000000000n`

Reply:
Updated the code pattern to make it similar to `toBigInt`